smu.c

```diff
@@ -243,6 +244,14 @@ dolineprefix(const char *begin, const char *end, int newblock) {
 			continue;
 		if (*begin == '\n')
 			fputc('\n', stdout);
+
+		/* All line prefixes add a block element. These are not allowed
+		 * inside paragraphs, so we must end the paragraph first. */
+		if (in_paragraph) {
+			fputs("</p>\n", stdout);
+			in_paragraph = 0;
```

Comment:
This block is repeated 3 times. IMO that's a good candidate for a function, `end_paragraph()` perhaps.